### PR TITLE
Test clang with and without optimisations

### PR DESCRIPTION
**Pull request by NWilson**

### CODE REVIEW

**Comment by NWilson** [comment]
The build failure here is due to #558

**Review by carenas — CHANGES_REQUESTED** [review]
it might be better to implement this test as part of another job that might also test other things in parallel that aren''t currently covered.

for example as part of a windows build that uses the alternative clang compiler that comes with Visual Studio and that might be updated in the future to build for ARM64

File: .github/workflows/dev.yml [hunk 1]
```diff
@@ -52,7 +55,7 @@ jobs:
         run: ./autogen.sh
 
       - name: Configure
-        run: ./configure CC='clang -fsanitize=undefined,address,integer -fno-sanitize-recover=undefined,integer -fno-sanitize=unsigned-integer-overflow,unsigned-shift-base,function' CPPFLAGS='-Wall -Wextra -Werror -Wno-error=unused-but-set-parameter -Wno-error=deprecated-declarations -Wno-error=incompatible-library-redeclaration -Wno-error=incompatible-pointer-types-discards-qualifiers' --enable-jit --enable-pcre2-16 --enable-pcre2-32 --enable-debug --with-link-size=3
+        run: ./configure CC='clang' CFLAGS='${{ matrix.opt }} -fsanitize=undefined,address,integer -fno-sanitize-recover=undefined,integer -fno-sanitize=unsigned-integer-overflow,unsigned-shift-base,function' CPPFLAGS='-Wall -Wextra -Werror -Wno-error=unused-but-set-parameter -Wno-error=deprecated-declarations -Wno-error=incompatible-library-redeclaration -Wno-error=incompatible-pointer-types-discards-qualifiers' --enable-jit --enable-pcre2-16 --enable-pcre2-32 --enable-debug --with-link-size=3
```
[the remark this answers is not in the document]
Comment (NWilson) [reply .github/workflows/dev.yml]: Good point, done.

**Comment by zherczeg** [comment]
Please update this patch to master.
After it is landed, the next step for me is #553

**Comment by NWilson** [comment]
Grr, there's still a build problem, even after your fix. I'll try and work it out as quick as possible.

It may be that running `clang -fsanitize -O0` simply isn't sensible.

**Comment by carenas** [comment]
> It may be that running `clang -fsanitize -O0` simply isn't sensible

yes, whichs is why I suggested doing it on windows instead (without. the sanitize part, of course)

**Comment by NWilson** [comment]
@zherczeg I've fixed it; the next build should pass, and will be OK to merge.

**Comment by zherczeg** [comment]
Ok. Lets wait the CI then.

**Pull request merged**